pkg/vtpm/src/server.cpp

```diff
@@ -499,14 +519,21 @@ serveClient(int sock, sockaddr_in clientAddr)
         cerr << "recv() from " << inet_ntoa(clientAddr.sin_addr)
              << " failed with " << rc << std::endl;
     } else if (recvBytes == CODED_STRM_HDR_LEN) {
-        if (handleRequest(sock, readHdr(hdrBuf)) != 0) {
-            cerr << "Failure processing the request" << std::endl;
+        google::protobuf::uint32 payloadSize = readHdr(hdrBuf);
+        if (payloadSize > 0 && payloadSize <= (MAX_PAYLOAD_SIZE - CODED_STRM_HDR_LEN)) {
+            if (handleRequest(sock, payloadSize) != 0) {
+                cerr << "Failure processing the request" << std::endl;
+            }
+        }
+        else {
+            cerr << "Failure processing the request, too large (maybe adjust MAX_PAYLOAD_SIZE)" << std::endl;
```

Comment:
Would it be useful to lig the received size here?

Reply:
done.